src/file.js

```diff
@@ -2071,21 +2129,38 @@ File.prototype.request = function(reqOpts, callback) {
 };
 
 /**
- * The Storage API allows you to use a custom key for server-side encryption.
  * This method allows you to update the encryption key associated with this
  * file.
  *
  * @see [Customer-supplied Encryption Keys]{@link https://cloud.google.com/storage/docs/encryption#customer-supplied}
  *
- * @param {string|buffer} encryptionKey An AES-256 encryption key.
+ * @param {string|buffer} key An AES-256 encryption key or Cloud KMS key name.
+ *     If a Cloud KMS key name, you must specify an options object to indicate
+ *     the new key is a cloud KMS key name.
+ * @param {object} [options] Configuration object.
+ * @param {boolean} [options.kms=false] Specify `true` if key is a Cloud KMS
+ *     key name.
  * @returns {File}
  *
  * @example <caption>include:samples/encryption.js</caption>
  * region_tag:storage_rotate_encryption_key
  * Example of rotating the encryption key for this file:
  */
-File.prototype.rotateEncryptionKey = function(encryptionKey, callback) {
-  var newFile = this.bucket.file(this.id, {encryptionKey});
+File.prototype.rotateEncryptionKey = function(key, options, callback) {
```

Comment:
This is not a great design here, so opinions welcome. The problem is that `rotateEncryptionKey` never expected that there would be two types of keys. To support a `kms` key, I created an `options` object, which has to specify `options.kms = true`.